lib/route.js

```diff
@@ -112,6 +112,7 @@ function buildRouting (options) {
     options = Object.assign({}, options, {
       method,
       url,
+      path: url,
```

Comment:
Why this addition?

Reply:
the options object for the `HEAD` route is created by merging a copy of the option object of the `GET` route with the additional properties listed above, the `url` passed when creating the `HEAD` route is an empty string which is not truthy so the `path` string from the option object is instead used which causes fastify to concatenate the prefix with the already prefixed path of the `GET` route to create the `HEAD` route path.

setting `path` property to the same value as the `url` that was passed fixed the issue.